src/Routes/routes.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { ReactElement } from "react";
+import { Routes, Route, Navigate } from "react-router-dom";
+import { LoginPage } from "../pages/login";
+import { HomePage } from "../pages/home";
+
+interface ChildrenTypes {
+  children: ReactElement;
+}
+
+const Private = ({ children }: ChildrenTypes) => {
```

Comment:
```suggestion
const AuthGuard: React.FC<PropsWithChildren> = ({ children }) => {
```
duas coisas aqui
1. esse padrão, que faz alguma verificação antes de redirecionar o usuário para a tela, é normalmente chamado de `guard` ([exemplo](https://jwt-auth.readthedocs.io/en/develop/auth-guard/)). Como vc tá verificando se o usuário está autenticado, `AuthGuard` seria um bom nome 😅 
2. O React já tem algo parecido com o `ChildrenTypes` que vc criou, chamado `PropsWithChildren`. Mostrei um exemplo de como usar acima ⬆️

Reply:
Opa! Valeu demais Beto! Não conhecia essa parte de ```guard```, vou dar uma lida com calma. E o ```PropsWithChildren``` também é um adianto haha

Obrigado! Corrigido